akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -131,13 +140,13 @@ object ShardRegion {
 
     override def shardId(message: Any): String = {
       val id = message match {
-        case ShardRegion.StartEntity(id) ⇒ id
-        case _                           ⇒ entityId(message)
+        case ShardRegion.StartEntity(entityId) ⇒ entityId
+        case _                                 ⇒ entityId(message)
       }
       // It would be better to have abs(id.hashCode % maxNumberOfShards), see issue #25034
       // but to avoid getting different values when rolling upgrade we keep the old way,
       // and it doesn't have any serious consequences
```

Comment:
This comment should be moved to `HashCodeMessageExtractor.shardId` and the implementation must unfortunately not be changed, due to reason stated in the comment.

Now we will use the same slightly wrong implementation for Typed too, but that is probably a good choice in case Typed and Untyped sharding are used together.

Reply:
@patriknw @johanandren just clarifying, I should close this and we leave as is?

Reply:
I think aligning the weird behavior is good, so push this PR across the line. (I was just saying that's probably why it was how it was.)

Reply:
👍 Thanks @johanandren